src/CoreText/CTLine.cs

```diff
@@ -238,37 +204,32 @@ public nfloat GetOffsetForStringIndex (nint charIndex)
 		[iOS (9,0)][Mac (10,11)]
 #endif
 		[DllImport (Constants.CoreTextLibrary)]
-		static unsafe extern void CTLineEnumerateCaretOffsets (IntPtr line, BlockLiteral *blockEnumerator);
+		static extern void CTLineEnumerateCaretOffsets (IntPtr line, ref BlockLiteral blockEnumerator);
 
 		static unsafe readonly CaretEdgeEnumeratorProxy static_enumerate = TrampolineEnumerate;
 
 		[MonoPInvokeCallback (typeof (CaretEdgeEnumeratorProxy))]
 		static unsafe void TrampolineEnumerate (IntPtr blockPtr, double offset, nint charIndex, bool leadingEdge, ref bool stop)
 		{
-			var block = (BlockLiteral *) blockPtr;
-                        var del = (CaretEdgeEnumerator) (block->Target);
-                        if (del != null)
-                                del (offset, charIndex, leadingEdge, ref stop);
- 		}
-		
+			var block = (BlockLiteral*) blockPtr;
+			var del = (CaretEdgeEnumerator) (block->Target);
```

Comment:
Could we have used as and then do it in the if statement?

Reply:
I simplified it a bit differently in the end.